File: android/src/oldarch/com/zoontek/rnlocalize/RNLocalizeModule.java
package com.zoontek.rnlocalize;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.ReadableArray;
import com.facebook.react.bridge.WritableArray;
import com.facebook.react.bridge.WritableMap;
import com.facebook.react.module.annotations.ReactModule;

import java.util.ArrayList;
import java.util.List;

@ReactModule(name = RNLocalizeModuleImpl.NAME)
public class RNLocalizeModule extends ReactContextBaseJavaModule {
  private final RNLocalizeModuleImpl moduleImpl;

  public RNLocalizeModule(ReactApplicationContext reactContext) {
    super(reactContext);
    this.moduleImpl = new RNLocalizeModuleImpl(reactContext);
  }

  @NonNull
  @Override
  public String getName() {
    return RNLocalizeModuleImpl.NAME;
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  public String getCalendar() {
    return moduleImpl.getCalendar();
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  public String getCountry() {
    return moduleImpl.getCountry();
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  public WritableArray getCurrencies() {
    return moduleImpl.getCurrencies();
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  public WritableArray getLocales() {
    return moduleImpl.getLocales();
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  public WritableMap getNumberFormatSettings() {
    return moduleImpl.getNumberFormatSettings();
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  public String getTemperatureUnit() {
    return moduleImpl.getTemperatureUnit();
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  public String getTimeZone() {
    return moduleImpl.getTimeZone();
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  public boolean uses24HourClock() {
    return moduleImpl.uses24HourClock();
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  public boolean usesMetricSystem() {
    return moduleImpl.usesMetricSystem();
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  public Boolean usesAutoDateAndTime() {
    return moduleImpl.usesAutoDateAndTime();
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  public Boolean usesAutoTimeZone() {
    return moduleImpl.usesAutoTimeZone();
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  public void setApplicationLocales(ReadableArray languageTagsArray) {
    moduleImpl.setApplicationLocales(languageTagsArray);
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  public WritableArray getApplicationLocales() {
    return moduleImpl.getApplicationLocales();
  }

  @ReactMethod
  public void addListener(String eventName) {}

  @ReactMethod
  public void removeListeners(double count) {}
}

File: android/src/oldarch/com/zoontek/rnlocalize/RNLocalizeModule.kt
package com.zoontek.rnlocalize

import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.WritableArray
import com.facebook.react.bridge.WritableMap
import com.facebook.react.module.annotations.ReactModule

@ReactModule(name = RNLocalizeModuleImpl.NAME)
class RNLocalizeModule(reactContext: ReactApplicationContext) :
  ReactContextBaseJavaModule(reactContext) {

  override fun getName(): String {
    return RNLocalizeModuleImpl.NAME
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  fun getCalendar(): String {
    return RNLocalizeModuleImpl.getCalendar()
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  fun getCountry(): String {
    return RNLocalizeModuleImpl.getCountry(reactApplicationContext)
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  fun getCurrencies(): WritableArray {
    return RNLocalizeModuleImpl.getCurrencies(reactApplicationContext)
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  fun getLocales(): WritableArray {
    return RNLocalizeModuleImpl.getLocales(reactApplicationContext)
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  fun getNumberFormatSettings(): WritableMap {
    return RNLocalizeModuleImpl.getNumberFormatSettings(reactApplicationContext)
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  fun getTemperatureUnit(): String {
    return RNLocalizeModuleImpl.getTemperatureUnit(reactApplicationContext)
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  fun getTimeZone(): String {
    return RNLocalizeModuleImpl.getTimeZone()
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  fun uses24HourClock(): Boolean {
    return RNLocalizeModuleImpl.uses24HourClock(reactApplicationContext)
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  fun usesMetricSystem(): Boolean {
    return RNLocalizeModuleImpl.usesMetricSystem(reactApplicationContext)
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  fun usesAutoDateAndTime(): Boolean {
    return RNLocalizeModuleImpl.usesAutoDateAndTime(reactApplicationContext)
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  fun usesAutoTimeZone(): Boolean {
    return RNLocalizeModuleImpl.usesAutoTimeZone(reactApplicationContext)
  }

  @ReactMethod
  fun openAppLanguageSettings(promise: Promise) {
    RNLocalizeModuleImpl.openAppLanguageSettings(reactApplicationContext, promise)
  }
}
